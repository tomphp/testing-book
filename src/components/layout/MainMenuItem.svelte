<script>
  export let link;
  export let title;
</script>

<li class="mr-6 font-bold text-center hover:bg-slate-200 hover:text-emerald-500 p-2">
  <a href="{link}" class="text-inherit">{title}</a>
</li>
