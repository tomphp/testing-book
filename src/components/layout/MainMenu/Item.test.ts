import {render} from "@testing-library/svelte";
import MainMenuItem from "./Item.svelte";

test("menu item with link", () => {
  const {container} = render(MainMenuItem, { props: { link: "/foo", title: "Foo" } });
  expect(container).toMatchSnapshot();
})

test("menu item without", () => {
  const {container} = render(MainMenuItem, { props: { title: "Foo" } });
  expect(container).toMatchSnapshot();
})
