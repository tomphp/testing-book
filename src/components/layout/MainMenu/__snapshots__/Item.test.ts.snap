// Vitest Snapshot v1

exports[`menu item with link 1`] = `
<body>
  <div>
    <li
      class="mr-0 font-bold text-center p-2"
    >
      <div
        class="flex justify-center"
      >
        <div>
          <div
            class="relative"
            data-te-dropdown-ref=""
          >
            <a
              aria-expanded="false"
              class="flex text-sm items-center whitespace-nowrap rounded bg-primary px-6 pt-2.5 pb-2 font-medium uppercase leading-normal text-white shadow-[0_4px_9px_-4px_#3b71ca] transition duration-150 ease-in-out hover:bg-white hover:text-emerald-500 hover:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:bg-primary-600 focus:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:outline-none focus:ring-0 active:bg-primary-700 active:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] motion-reduce:transition-none"
              data-te-dropdown-toggle-ref=""
              data-te-ripple-color="light"
              data-te-ripple-init=""
              href="/foo"
              id="dropdownMenuButton2"
              type="button"
            >
              Foo
               
            </a>
             
          </div>
        </div>
      </div>
    </li>
    <!--&lt;Item&gt;-->
  </div>
</body>
`;

exports[`menu item without 1`] = `
<body>
  <div>
    <li
      class="mr-0 font-bold text-center p-2"
    >
      <div
        class="flex justify-center"
      >
        <div>
          <div
            class="relative"
            data-te-dropdown-ref=""
          >
            <a
              aria-expanded="false"
              class="flex text-sm items-center whitespace-nowrap rounded bg-primary px-6 pt-2.5 pb-2 font-medium uppercase leading-normal text-white shadow-[0_4px_9px_-4px_#3b71ca] transition duration-150 ease-in-out hover:bg-white hover:text-emerald-500 hover:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:bg-primary-600 focus:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:outline-none focus:ring-0 active:bg-primary-700 active:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] motion-reduce:transition-none"
              data-te-dropdown-toggle-ref=""
              data-te-ripple-color="light"
              data-te-ripple-init=""
              id="dropdownMenuButton2"
              type="button"
            >
              Foo
               
            </a>
             
          </div>
        </div>
      </div>
    </li>
    <!--&lt;Item&gt;-->
  </div>
</body>
`;
